bootstrap/kubeadm/api/v1beta1/kubeadm_types.go

```diff
@@ -520,6 +526,7 @@ type Discovery struct {
 type BootstrapTokenDiscovery struct {
 	// token is a token used to validate cluster information
 	// fetched from the control-plane.
+	// +required
 	Token string `json:"token"`
```

Comment:
I know I'm a bit late, but is the Token field here really required? We were using a KubeadmConfigTemplate for quite some time with the token omitted until a bump of the dependencies broke this workflow.

Reply:
This change didn't make the field required. controller-gen already considered the field required before. This is also why this PR didn't change the CRD YAML at all.

Wondering how this chnage broke your workflow :)

Reply:
I think what might have broken your flow is that a subsequen PR added MinLength validation

Reply:
Then this PR made the field optional though: https://github.com/kubernetes-sigs/cluster-api/pull/12107

(not released yet)

Reply:
Thanks, really appreciate your quick response!